routes/chat.js

```diff
@@ -0,0 +1,123 @@
+const router = require("./comment");
+const mongoose = require("mongoose");
+const User = require("../models/user");
+const socketIO = require("socket.io");
+const users = {};
+
+router.post("/getFriendsChat", function(request, result) {
+    var _id = request.fields._id;
+    database.collection("users").findOne({
+        "email" : request.fields.email
+    }, function(error, user) {
+        if(user == null) {
+            result.json({
+                "status": "error",
+                "message": "User has been logged out. Please Login again"
```

Comment:
We cannot infer here that the user is logged out.  This message should be to enter a valid user email to send a message.